docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,457 @@
+`ConfiguredGraphFactory` static methods. For example, you may use
+`ConfiguredGraphFactory.removeTemplateConfiguration` instead of
+`ConfiguredGraphFactory.getInstance().removeTemplateConfiguration()`.
+
+[[graph-configurations]]
+==== Graph Configurations
+
+The `ConfigurationManagementGraph` singleton allows you to create
+configurations used to open specific graphs, referenced by the
+`graph.graphname` property. For example:
+
+[source, gremlin]
+----
+Map<String, Object> map = new HashMap<String, Object>();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+map.put("graph.graphname", "graph1");
+ConfiguredGraphFactory.createConfiguration(new MapConfiguration(map));
+----
+
+Then you could access this graph on any JanusGraph node using:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.open("graph1");
+----
+
+[[template-configuration]]
+==== Template Configuration
+
+The `ConfigurationManagementGraph` also allows you to create one
+template configuration, which you can use to create many graphs using
+the same configuration template. For example:
+
+[source, gremlin]
+----
+Map<String, Object> map = new HashMap<String, Object>();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+ConfiguredGraphFactory.createTemplateConfiguration(new MapConfiguration(map));
+----
+
+After doing this, you can create graphs using the template
+configuration:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.create("graph2");
+----
+
+This method will first create a new configuration for "graph2" by
+copying over all the properties associated with the template
+configuration and storing it on a configuration for this specific graph.
+This means that this graph can be accessed in, on any JanusGraph node,
+in the future by doing:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.open("graph2");
+----
+
+[[updating-configurations]]
+==== Updating Configurations
+
+All interactions with both the `JanusGraphFactory` and the
+`ConfiguredGraphFactory` that interact with configurations that define
+the property `graph.graphname` go through the `JanusGraphManager` which
+keeps track of graph references created on the given JVM. Think of it as
+a graph cache. For this reason:
+
+IMPORTANT: Any updates to a configuration are not guaranteed to take effect until
+you remove the graph in question on every janusgraph node in your
+cluster.
+
+You can do so by calling:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.close("graph2");
+----
+
+Since graphs created using the template configuration first create a
+configuration for that graph in question using a copy and create method,
+this means that:
+
+IMPORTANT: Any updates to a specific graph created using the template
+configuration are not guaranteed to take effect on the specific graph
+until:
+
+1. The relevant configuration is removed: `ConfiguredGraphFactory.removeConfiguration("graph2");`
```

Comment:
This ordered list doesn't appear inside the `IMPORTANT` box, but rather outside, which looks a bit funny:
https://pluradj.github.io/janusgraph.org/latest/configuredgraphfactory.html